jetty-util/src/test/java/org/eclipse/jetty/util/HostPortTest.java

```diff
@@ -77,7 +79,21 @@ public static Stream<Arguments> invalidAuthorityProvider()
             "host:-80", // host + invalid port
             "127.0.0.1:-80", // ipv4 + invalid port
             "[0::0::0::0::1]:-80", // ipv6 + invalid port
```

Comment:
Can you also test `[0::0::0::0::1].80` and make sure it does not parse 80 as the port.

Reply:
Added.

Reply:
Yeah but I think with the current code it will be parsed as host `[0::0::0::0::1]` and port 80.    I think in this case it should be host `[0::0::0::0::1].80` and port -1